Waveform module changes for precessing search

spxiwh opened the pull request:
These are the changes to the waveform module needed for the precessing search:

 * We add a two-polarization waveform generation routine, allowing a template to return both h+ and hx (using the same ideas and methods as the single-polarization call)
 * We add in length estimates for IMRPhenomPv2 and SpinTaylorF2. Currently this is not great (as it uses an aligned-spin estimator), but can be improved in the future.
 * We add a fall-back hack for SpinTaylorF2, such that if the system is aligned-spin, *and* spin2z is non-zero the approximant will fall back to TaylorF2. This allows us to have a template bank containing aligned-spin spin2!=0 waveforms and precessing spin2=0 waveforms with the [Spin]TaylorF2 approximant.

cdcapano commented on pycbc/waveform/waveform.py:
```diff
@@ -88,6 +88,15 @@ def _lalsim_fd_waveform(**p):
+            # Fall back to standard F2
+            p['approximant'] = 'TaylorF2'
+
+
```
Can you move this to inside get_two_pol_waveform_filter? I'm a bit wary of allowing this to silently happen even when calling get_fd_waveform, which should just be a thin wrapper around the underlying waveform generators.

spxiwh replied:
This is now a dedicated approximant (SpinTaylorF2_SWAPPER[_INTERP]) as suggested.

cdcapano commented on pycbc/waveform/waveform.py:
```diff
@@ -705,6 +716,66 @@ def td_waveform_to_fd_waveform(waveform, out=None, length=None,
 
+def get_two_pol_waveform_filter(outplus, outcross, template, **kwargs):
+    """Return a frequency domain waveform filter for the specified approximant
+    """
```
Please make the doc string more descriptive. Also, stick a note here about the SpinTaylorF2 --> TaylorF2 when the spins are anti-aligned.

spxiwh replied:
Doc string updated

spxiwh commented:
Hi @cdcapano , please see updated patch. As well as addressing the concerns, I also make some edits to the INTERP approximant so that it does not try to INTERP things it does not work with (ie. it doesn't work with SPAtmplt at the moment).

I also fix the SingleSpin - > EffectiveSpin renaming in the SEOBNR ROMs. Now the waveform test passes .... I wonder how we did not know it was failing before this? Is the Travis not testing this.
